call_esm_pretrained_cafa5.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Dict
+import torch
+from transformers import AutoTokenizer, EsmForSequenceClassification
+
+from constants import TRAIN_FASTA_FPATH, TRAIN_TERMS_FPATH, OBO_FPATH, MODEL_ID, MINIMUM_PROBABILITY
+
+
+class ProteinFunctionPredictor:
+    def __init__(self, model, tokenizer, unique_terms: List[str], go_terms: Dict[str, dict]):
+        self.model = model
+        self.tokenizer = tokenizer
+        self.unique_terms = unique_terms
+        self.go_terms = go_terms
+
+    @classmethod
+    def from_files(cls, model_id: str = MODEL_ID, tsv_fpath: str = TRAIN_TERMS_FPATH, obo_fpath: str = OBO_FPATH):
+        tsv_data = cls.parse_tsv_file(tsv_fpath)
+        unique_terms = list(set(term for terms in tsv_data.values() for term in terms))
+
+        go_terms = cls.parse_obo_file(obo_fpath) 
+        loaded_model = EsmForSequenceClassification.from_pretrained(model_id)
+        loaded_tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+        return cls(loaded_model, loaded_tokenizer, unique_terms, go_terms)
+
+    @staticmethod
+    def parse_tsv_file(file_path: str) -> Dict[str, List[str]]:
+        tsv_data = {}
+        with open(file_path, 'r', encoding='utf-8') as f:
+            for line in f:
+                parts = line.strip().split("\t")
+                tsv_data[parts[0]] = parts[1:]
+        return tsv_data
+
+    @staticmethod
+    def parse_obo_file(file_path: str) -> List[dict]:
+        with open(file_path, 'r', encoding='utf-8') as f:
+            data = f.read().split("[Term]")
+
+        go_terms_by_id = {}
+        for entry in data[1:]:
+            lines = entry.strip().split("\n")
+            term = {}
+            for line in lines:
+                if line.startswith("id:"):
+                    term["id"] = line.split("id:")[1].strip()
+                elif line.startswith("name:"):
+                    term["name"] = line.split("name:")[1].strip()
+                elif line.startswith("namespace:"):
+                    term["namespace"] = line.split("namespace:")[1].strip()
+                elif line.startswith("def:"):
+                    term["definition"] = line.split("def:")[1].split('"')[1]
+            go_terms_by_id[term["id"]] = term
+        return go_terms_by_id
+
+    # 3. The predict_protein_function function
+    def classify(self, sequence: str) -> List[str]:
+        inputs = self.tokenizer(sequence, return_tensors="pt", padding=True, truncation=True, max_length=1022)
```

Comment:
I don't think padding is necessary for single sequence samples - it's not hurting either. Where it would come into play is if you had a batch of sequences with different lengths.